test/testcne/metrics_test.c

```diff
@@ -0,0 +1,97 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ * Copyright (c) 2022 Intel Corporation
+ */
+
+// IWYU pragma: no_include <bits/getopt_core.h>
+
+#include <stdio.h>           // for NULL, EOF
+#include <metrics.h>         // for metrics_register
+#include <tst_info.h>        // for tst_error, tst_end, tst_start, TST_FAILED
+
+#include "metrics_test.h"
+
+static int
+test_cb(metrics_client_t *client, const char *cmd __cne_unused, const char *params __cne_unused)
+{
+    cne_printf("Test Callback\n");
+    metrics_append(client, "\"name\":\"Metrics Test\"");
+    return 0;
+}
+
+static int
+metrics_test(void)
+{
+    const char *c;
+    int ret                  = 0;
+    metrics_client_t *client = calloc(1, sizeof(metrics_client_t));
+    client->cmd              = "/test";
+    client->params           = "params";
+    lport_stats_t stats      = {0};
+    char lpname[]            = "test_lport";
+
+    ret = metrics_init(NULL);
+    if (ret < 0) {
+        tst_error("Unable to initilize metrics library\n");
+        goto err;
```

Comment:
if we goto err; here , metrics_destroy() will be called. If metrics_init() fails, we don't want metrics_destroy() to be called, right?